packages/utils/src/lib/reports/generate-md-report.ts

```diff
@@ -26,19 +30,25 @@ export function auditDetailsAuditValue({
   )} (score: ${formatReportScore(score)})`;
 }
 
+function hasCategories(
+  report: ScoredReport,
+): report is ScoredReport & { categories: ScoredCategoryConfig[] } {
```

Comment:
It would be safer to derive categories type from `ScoredReport` (TS doesn't validate if the type predicate matches the implementation):

```suggestion
): report is ScoredReport & Required<Pick<ScoredReport, 'categories'>> {
```